### DISCUSSION TITLE
Stat with charts color bug

### DISCUSSION THREAD

**Discussion by aldozumaran** [filamentphp/filament#10637] [Help]
i have this widget, every stat with different color
color does not work!
charColor is the same 
```
protected function getStats(): array
    {
        return [
            Stat::make('Unique views', '192.1k')
                ->description('32k increase')
                ->descriptionIcon('heroicon-m-arrow-trending-up')
                ->color('success')
                ->descriptionColor('success')
                ->chartColor('success')
                ->chart([2,6,1,7,9,3,5,7,1]),
            Stat::make('Bounce rate', '21%')
                ->description('7% increase')
                ->descriptionIcon('heroicon-m-arrow-trending-down')
                ->color('danger')
                ->descriptionColor('danger')
                ->chartColor('danger')
                ->chart([1,2,1,8,6,5,3,9,10]),
            Stat::make('Average time on page', '3:12')
                ->description('3% increase')
                ->descriptionIcon('heroicon-m-arrow-trending-up')
                ->color('primary')
                ->descriptionColor('primary')
                ->chartColor('primary')
                ->chart([6,9,3,6,12,4,7,12,7]),
        ];
    }
```

![image](https://github.com/filamentphp/filament/assets/3183704/beb16bf3-9616-48b0-968b-0b7a60431fa3)

**Comment by leandrocfe** [comment]
It should work with `->color('success')`
What is the Filament version?

**Reply by aldozumaran** [reply]
filament v3

**Reply by leandrocfe** [reply]
3.?.?

**Reply by aldozumaran** [reply]
3.1.33

**Reply by leandrocfe** [reply]
Did you customize something in the project? Did you do something in the AppServiceProvider?

**Reply by devidsign** [reply]
i'm using a custom dashboard

**Reply by devidsign** [reply]
works fine using the default dashboard

**Reply by leandrocfe** [reply]
Some custom code you added in the provider

**Comment by DawidMazur** [comment]
I have same issue on defaut dashboard, but in:

            Stat::make('profits', MoneyHelper::getFormatedMoney($sales['sum']))
                ->label('Przychody')
                ->description( 'Przychody wzrosły o ' . 57.56 . '%, w porównaniu z ostatnim miesiącem.')
                ->chart(Collection::make($sales['sales_per_day'])->pluck('sum')->toArray())
                ->chartColor('success')
                ->color('success')
                ,
            Stat::make('cost', MoneyHelper::getFormatedMoney($costs['sum']))
                ->label(__('Koszty'))
                ->description(__('W przeliczeniu na dziś'))
                ->chart(Collection::make($costs['costs_per_day'])->pluck('sum')->toArray())
                ->chartColor('danger')
                ->color('danger'),
                
                
No custom code in this area (chart is generated correctly, and data.
![image](https://github.com/filamentphp/filament/assets/50441054/6f69a7fc-a5bf-46e3-8ccc-35e2bec17eef)

**Reply by leandrocfe** [reply]
Filament version? 3.?.?

**Comment by neohades** [comment]
I'm using Filament ^3.2.

What I've found out is that when You set the chartColor to anything, the color is not changing properly, it changes to "custom" every time.

In the example below screenshot, You can find (1) which is the chartColor that I've set (primary), and in the (2) there is `text-custom-50` and `text-custom-500`, and in there should be the proper color set by chartColor, so there should be `text-primary-50` and `text-primary-500`.

![Selection_061](https://github.com/filamentphp/filament/assets/2737563/75d6ad84-c721-427b-bc48-04ba538a6c25)

As You may see, I've managed to change chart color, but only thatnks to CSS overwrite. I've changed the CSS `color` props in the below classes, and it worked.
`.fi-wi-stats-overview-stat-chart .text-custom-50` and `.fi-wi-stats-overview-stat-chart .text-custom-500` 

Nevertheless, the chartColor does not work properly, something is not passed as it should be.

As i found out, the logic that stand behing this is located in the core file filament/widgets/resources/views/stats-overview-widget/stat.blade.php between lines 137-155, there is the "match" condition upon 'chartColor' that actually does not pick anything else than gray or custom. I would propose a change, but I'm not quite sure why it is the way it is :)

**Reply by neohades** [reply]
ok, I found out that the "custom" has the proper logic behind it, and my problem was related more to Tailwind config and less the Filament logic itself.

however the below comment is partialy right, somehow the vars `--c-50:var(--primary-50);--c-400:var(--primary-400);--c-500:var(--primary-500);` are being overwritten, but it is not a constant problem, just happens sometimes. still the CSS "hack" from the above comment works and the colors are not beeing overwritten this way

**Reply by nbankzy** [reply]
I ran into a similar situation using a Wizard on a livewire component on the front end of the website. Headings are inheriting primary (blue color) correctly but the NEXT buttons are displaying as the default filament orange. I found 2 solutions to overriding it.

1) Adding a color.custom property in my tailwind.config.js
```
theme: {
    extend: {
        colors: {
            custom: {
                '50': '#f0f7ff',
                '100': '#dfeeff',
                '200': '#b9defe',
                '300': '#7bc5fe',
                '400': '#34a7fc',
                '500': '#0a8ded',
                '600': '#006dcb',
                '700': '#005bac',
                '800': '#054b87',
                '900': '#0a3e70',
                '950': '#07274a',
            },
        }
    }
}
```

2) [Registering a non custom color in tailwind](https://filamentphp.com/docs/3.x/support/colors#using-a-non-tailwind-color). In my AdminPanelProvider (can be done in any provider) add the following.
```
public function boot(): void {
        // Register custom colours to be used outside of the panel
        \Filament\Support\Facades\FilamentColor::register([
            'primary' => '#005bac',
        ]);
}
```

The styles that are added inline on the button `--c-400:var(--primary-400);--c-500:var(--primary-500);--c-600:var(--primary-600);` are now pulling the newly defined primary values.
From looking at what else that's changed color, it seems that buttons are the most effected.

**Comment by WebScoutAgency** [comment]
I see that this isn't working even on filament demo:
Firstly was this:
![image](https://github.com/filamentphp/filament/assets/121947104/5cf00eec-3a02-4d81-8a38-bbbe800fa432)
After 5 second (problably interval refresh) this:
![image](https://github.com/filamentphp/filament/assets/121947104/97bb3687-e18a-475e-a89d-1a9d24d994ed)

**Reply by WebScoutAgency** [reply]
From here:
https://demo.filamentphp.com/

**Comment by ryanmortier** [comment]
I created an issue here: https://github.com/filamentphp/filament/issues/13518
